src/realm/util/load_file.cpp

```diff
@@ -13,19 +13,10 @@ std::string util::load_file(const std::string& path)
     for (;;) {
         std::size_t min_extra_capacity = 256;
         buffer.reserve_extra(used_size, min_extra_capacity);                             // Throws
-        std::size_t n = file.read(buffer.data() + used_size, buffer.size() - used_size); // Throws
+        std::size_t n = file.read(used_size, buffer.data() + used_size, buffer.size() - used_size); // Throws
         if (n == 0)
             break;
         used_size += n;
     }
     return std::string(buffer.data(), used_size); // Throws
 }
-
-
-std::string util::load_file_and_chomp(const std::string& path)
```

Comment:
Could you move the simple deletions/cleanups like this one into a separate PR, a lot of the file/mapping stuff is probably too interconnected to be worth extracting, but it would be nice to reduce the number of non-encryption related clean-ups here. (These are all great though!)

Reply:
There's a few steps in between, but this is actually connected to the encryption changes. Making `File::seek()` work for encrypted files was previously done with a global mutex which I wanted to kill. Rather than trying to figure out some locking scheme that would make seeking work I updated all of our uses of File to not rely on syncing and instead issue atomic read/write calls at specific offsets. For each thing I had to update for this I first checked if it was actually still used and just deleted it if not.

Ideally the File changes would all be a separate commit that goes before the encryption changes but they are pretty entangled, largely because of the map_flags thing that was passed around through every layer without ever being used for anything.